packages/apps/job-launcher/server/src/modules/job/job.service.ts

```diff
@@ -923,24 +925,96 @@ export class JobService {
       PaymentType.SLASH,
     );
     if (!slash?.length) {
-      const refund = await EscrowUtils.getCancellationRefund(
-        jobEntity.chainId,
-        jobEntity.escrowAddress!,
-      );
+      let refund: ICancellationRefund | null = null;
+      try {
+        refund = await EscrowUtils.getCancellationRefund(
+          jobEntity.chainId,
+          jobEntity.escrowAddress!,
+        );
+      } catch {
+        // Ignore error
+      }
 
-      if (!refund || !refund.amount) {
-        throw new ConflictError(ErrorJob.NoRefundFound);
+      let amount = 0n;
+
+      if (!refund) {
+        //Temp fix
+        amount = await this.getRefundAmount(
+          jobEntity.chainId,
+          jobEntity.escrowAddress!,
+          token.address,
+        );
+      } else {
+        if (!refund.amount) {
+          throw new ConflictError(ErrorJob.NoRefundFound);
+        }
+        amount = refund.amount;
       }
 
-      await this.paymentService.createRefundPayment({
-        refundAmount: Number(ethers.formatUnits(refund.amount, token.decimals)),
-        refundCurrency: jobEntity.token,
-        userId: jobEntity.userId,
-        jobId: jobEntity.id,
-      });
+      if (amount > 0n) {
+        await this.paymentService.createRefundPayment({
+          refundAmount: Number(ethers.formatUnits(amount, token.decimals)),
+          refundCurrency: jobEntity.token,
+          userId: jobEntity.userId,
+          jobId: jobEntity.id,
+        });
+      }
     }
 
     jobEntity.status = JobStatus.CANCELED;
     await this.jobRepository.updateOne(jobEntity);
   }
+
+  public async getRefundAmount(
+    chainId: ChainId,
+    escrowAddress: string,
+    tokenAddress: string,
+  ): Promise<bigint> {
+    const signer = this.web3Service.getSigner(chainId);
+    const provider = signer.provider;
+    const contract: Escrow = Escrow__factory.connect(escrowAddress!, provider);
+    const fromBlock = 79278120; //This issue started at this block
+    const toBlock = 'latest';
+    const cancelledFilter = contract.filters?.Cancelled?.();
+    const cancelledLogs = await contract.queryFilter(
+      cancelledFilter,
+      fromBlock,
+      toBlock,
+    );
```

Comment:
You can just double-check contract status and:
- throw if it's not `cancelled`
- if it's `cancelled` - query `Transfer` events and then compare transfer data with launcher address

That would make implementation much simpler, because now it's hard to get why you need to filter cancellation and use it as ref logs